pkg/asset/machines/gcp/gcpmachines.go

```diff
@@ -0,0 +1,126 @@
+// Package gcp generates Machine objects for gcp.
+package gcp
+
+import (
+	"fmt"
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+// GenerateMachines returns manifests and runtime objects to provision the control plane (including bootstrap, if applicable) nodes using CAPI.
+func GenerateMachines(clusterID string, region string, subnets map[string]string, pool *types.MachinePool, role string, projectID string, userTags map[string]string, userLabels map[string]string) ([]*asset.RuntimeFile, error) {
+	if poolPlatform := pool.Platform.Name(); poolPlatform != gcptypes.Name {
+		return nil, fmt.Errorf("non-GCP machine-pool: %q", poolPlatform)
+	}
+	mpool := pool.Platform.GCP
+
+	total := int64(1)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+
+	// Make sure subnet is defined
+	if _, ok := subnets[role]; !ok {
+		return nil, fmt.Errorf("subnet is not defined for role %s", role)
+	}
+	subnet := subnets[role]
+
+	var osImage string
+	if mpool.OSImage != nil {
+		osImage = mpool.OSImage.Name
+	}
+
+	// Set SecureBoot
+	shieldedInstanceConfig := capg.GCPShieldedInstanceConfig{}
+	if mpool.SecureBoot == string(machineapi.SecureBootPolicyEnabled) {
+		shieldedInstanceConfig.SecureBoot = capg.SecureBootPolicyEnabled
+	}
+
+	// Set ServiceAccount
+	instanceServiceAccount := fmt.Sprintf("%s-%s@%s.iam.gserviceaccount.com", clusterID, role[0:1], projectID)
+	// TODO - create a different service account for an xpn installation similar to current implementation ?
+
+	var result []*asset.RuntimeFile
+	for idx := int64(0); idx < total; idx++ {
+		gcpMachine := &capg.GCPMachine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "GCPMachine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-%s-%d", clusterID, pool.Name, idx),
```

Comment:
Did this include the role originally too ?

Reply:
> Did this include the role originally too ?

Yes, for the name...
machineAPI:
```
kind: Machine
metadata:
  name: bobf-test-cluster-nr72m-master-0
```
vs Capi
```
kind: GCPMachine
metadata:
  name: bobf-test-cluster-nr72m-master-0

```